sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -125,6 +125,17 @@ public class ServiceBusProcessor : IAsyncDisposable
         /// successful execution of the operation; otherwise, false.</value>
         public virtual bool AutoCompleteMessages { get; }
 
+        /// <summary>
+        /// Gets or sets whether the messages bodies use pooled byte arrays underneath in case of binary data.
+        /// When this option is enabled <see cref="ServiceBusReceivedMessage"/> cannot be safely kept for longer than the
+        /// lifetime of the scope of the <see cref="ServiceBusProcessor.ProcessMessageAsync"/> or
+        /// the <see cref="ServiceBusProcessor.ProcessSessionMessageAsync"/> event handler delegate.
```

Comment:
Let's only reference the ProcessMessageAsync here and the session one in the session options.

Reply:
ok will fix later

Reply:
I will potentially wait with fixing this until all the discussions have settled so that I can do things in one sweep.

Reply:
Was a matter of minutes. done